datafusion/expr/src/expr.rs

```diff
@@ -970,6 +976,72 @@ impl GroupingSet {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, Debug, Default)]
+pub struct WildcardOptions {
```

Comment:
as a follow on PR, can we add some doc comments to this struct explaining why this structure is needed and how to interpret it?

It seems like the core rationale is that wildcards have different semantics depending on what part of the query they appear in?

Reply:
Sure, I added descriptions for the purpose and source of each option.